packages/react-native/ReactCommon/react/renderer/components/root/RootShadowNode.cpp

```diff
@@ -49,7 +49,9 @@ RootShadowNode::Unshared RootShadowNode::clone(
           /* .props = */ props,
       });
 
-  if (layoutConstraints != getConcreteProps().layoutConstraints) {
+  if (layoutConstraints != getConcreteProps().layoutConstraints ||
+      layoutContext.pointScaleFactor !=
+          getConcreteProps().layoutContext.pointScaleFactor) {
     newRootShadowNode->dirtyLayout();
   }
```

Comment:
We have a down-propagation/inheritance step, during commit, once we have the full shadow-tree constructed, where we propagate `pointScaleFactor` to every Yoga node. https://github.com/facebook/react-native/blob/d8739e1d37ceebb9643ca56d66a54aaca330665a/packages/react-native/ReactCommon/react/renderer/components/view/YogaLayoutableShadowNode.cpp#L462

`YGNodeSetConfig` will do a comparison of previous and current config to dirty node (though changing something on the actual config after-the-fact does not, which is an API bug). So this code has a defect where it is not dirtying underlying Yoga node, if pointScaleFactor changes.

We probably want to fixup the invalidation when setting pointScaleFactor (either there in RN, or in Yoga), instead of just invalidating the root node.

An aside, I know the code already does this, but this method of dirtying that Fabric uses is a smell (and Yoga's public API only allows dirtying nodes with measure functions which can be impacted by external changes). Yoga tries to manage dirtying in all other cases, when APIs are called which mutate style or config. `RootShadowNode::layoutIfNeeded` should normally be able to run, even if the node is not dirty, and work only happens if the passed in constraints have changed.